src/pages/workspaces/workspace/analysis/modals/ComputeModal.js

```diff
@@ -1756,16 +1756,21 @@ export const ComputeModalBase = ({
     ])
   }
 
-  const renderPersistentDiskSection = diskExists => {
+  const RenderPersistentDiskSection = ({ existingPersistentDisk }) => {
```

Comment:
Somewhat of a nit: since these are now components, it doesn't make sense to start them with "Render". I would suggest "PersistentDiskSection" and "PersistentDiskType".

Also, if you want to experiment with simplifying the input arg so it's just the boolean `existingPersistentDisk` instead of an object, I'm pretty sure that will work.